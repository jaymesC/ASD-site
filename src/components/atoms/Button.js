import React from 'react'
import styled from 'styled-components'

export default function Button({label}) {
  const Button = styled.button`
  background-color:#d12e1a;
  color:#000;
  width:50px;
  padding:20px;
  
  `
  return (
    <div>
      <Button></Button>
    </div>
  )
}
