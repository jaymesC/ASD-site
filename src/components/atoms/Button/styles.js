import styled from 'styled-components';

export const StyledButton = styled.button`
  background-color: #e64b38;
  transition: 0.5s;

  &:hover {
    background-color: #d12e1a;
  }

  color: #fff;
  width: 350px;
  padding-top: 22px;
  padding-bottom: 22px;
  border-radius: 30px;
  border: #ed6050;
  cursor: pointer;
  font-family: 'inter';
  font-size: 16px;
  font-weight: 500;
`;
