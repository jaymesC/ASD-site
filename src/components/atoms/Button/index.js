import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton } from './styles';

/**
 * @description this is a plain button component with custom styles for the app
 *
 * @param {object} props the props contains a children field that accepts
 * all renderable elements.
 * @return {JSX.Element}
 * @constructor
 */

/**
 * 
 * @param {*} props 
 * @returns 
 */
const Button = (props) => {
  return (
    <div>
      <StyledButton>{props.children}</StyledButton>
    </div>
  );
};

// Button props
Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default Button;
