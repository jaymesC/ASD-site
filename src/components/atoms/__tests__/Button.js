import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button>Default Starter</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
