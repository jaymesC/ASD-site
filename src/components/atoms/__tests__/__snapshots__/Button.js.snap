// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    className="sc-bdnxRM llhdio"
  >
    Default Starter
  </button>
</div>
`;
